NtupleProducer/plugins/bRegressionProducer.cc

```diff
@@ -0,0 +1,567 @@
+/*
+  Adapted from michelif/flashgg implementation:
+  https://github.com/michelif/flashgg/blob/bregProducer_90X/Taggers/plugins/flashggbRegressionProducer.cc
+
+ TODO:
+ - Find out which is the correct weight file
+ - Find out if/how to consider only b-jets (or b-reg applied to ALL jets?)
+*/
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "RecoJets/JetProducers/interface/PileupJetIdAlgo.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "PhysicsTools/TensorFlow/interface/TensorFlow.h"
+
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+#include "RecoVertex/VertexPrimitives/interface/ConvertToFromReco.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexState.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+
+#define debug 0
+
+using namespace std;
+using namespace edm;
+
+class bRegressionProducer : public EDProducer
+{
+
+public:
+    bRegressionProducer( const ParameterSet & );
+    ~bRegressionProducer(){};
+    void InitJet();
+    void SetNNVectorVar();
+    std::vector<float> EvaluateNN();
+private:
+    void produce( Event &, const EventSetup & ) override;
+
+    edm::InputTag inputTagJets_;
+    EDGetTokenT<View<pat::Jet> > jetToken_;
+    edm::EDGetTokenT<double> rhoToken_;      
+ 
+    FileInPath bRegressionWeightfile_;
+
+    edm::EDGetTokenT<std::vector<reco::Vertex>> srcVtx_;
+    edm::EDGetTokenT<edm::View<reco::VertexCompositePtrCandidate>> srcSV_;  
+
+    double y_mean_;
+    double y_std_;
+
+    tensorflow::Session* session;
+    std::vector<float> NNvectorVar_; 
+
+
+    //mva variables
+    float Jet_pt ;
+    float Jet_eta ;
+    float rho ;
+    float Jet_mt ;
+    float Jet_leadTrackPt ;
+    float Jet_leptonPtRel ;
+    float Jet_leptonDeltaR ;
+    float Jet_neHEF ;
+    float Jet_neEmEF ;
+    float Jet_vtxPt ;
+    float Jet_vtxMass ;
+    float Jet_vtx3dL ;
+    float Jet_vtxNtrk ;
+    float Jet_vtx3deL ;
+    float Jet_energyRing_dR0_em_Jet_e ;
+    float Jet_energyRing_dR1_em_Jet_e ;
+    float Jet_energyRing_dR2_em_Jet_e ;
+    float Jet_energyRing_dR3_em_Jet_e ;
+    float Jet_energyRing_dR4_em_Jet_e ;
```

Comment:
Same comments as before; merging all these dR rings into ```std::array```'s would simplify the code.

Reply:
I suppose the same argument somewhat holds here that the variables are appearing consistently in the same order due to the way `NNvectorVar_` has to be filled. If this is really bugging you I can do this change, but otherwise I think I'd rather leave this as it it